src/api/route-services/projects.js

```diff
@@ -243,6 +243,7 @@ class ProjectsService {
         const deletePromises = experiments.reduce((acc, experimentId) => {
           acc.push(experimentService.deleteExperiment(experimentId));
           acc.push(samplesService.deleteSamplesEntry(projectUuid, experimentId, sampleUuids));
+          acc.push(accessService.deleteExperiment(experimentId));
```

Comment:
isn't it better if `accessService.deleteExperiment(experimentId)` is getting called as part of `experimentService.deleteExperiment(experimentId)` inside the `deleteExperiment` function of the `ExperimentService` class in `api/route-services/experiment`?

My reasoning is that it is part of deleting an experiment and if we call `accessService.deleteExperiment` from within `experimentService.deleteExperiment`, there are less moving parts involved. In particular, there is lower chance of us just calling `experimentService.deleteExperiment` from somewhere and forgetting to also call `accessService.deleteExperiment`

Reply:
mmm I kind of agree, but then we should also move the `samplesService` delete inside the other one. This three calls will be removed and become a single one when we migrate into SQL anyway.

Reply:
ok, so if they will be removed as part of the data model migration, I agree it is not worth refactoring this